File: RepoTransBench/target_projects/Python/Go/stjordanis_betterprompt/public_tests/public_betterprompt_test.go
package public_tests

import (
	"math"
	"os"
	"testing"

	"betterprompt"
)

func equalFloatSlice(a, b []float64) bool {
	if len(a) != len(b) {
		return false
	}
	for i := range a {
		if math.Abs(a[i]-b[i]) > 1e-9 {
			return false
		}
	}
	return true
}
func sum(arr []float64) float64 {
	s := 0.0
	for _, v := range arr {
		s += v
	}
	return s
}

func setEnv(key, val string) (orig string, present bool) {
	orig, present = os.LookupEnv(key)
	os.Setenv(key, val)
	return
}

func TestPublicCallOpenAICustomModel(t *testing.T) {
	dummyLogprobs := []float64{-0.1, 2.5, -4.3}
	betterprompt.MockOpenAICompletionWithModel(dummyLogprobs, "public-model")
	res, err := betterprompt.CallOpenAI("sample prompt here", "anypublickey", "public-model")
	if err != nil {
		t.Errorf("Unexpected error: %v", err)
	}
	if !equalFloatSlice(res, dummyLogprobs) {
		t.Errorf("Expected %v, got %v", dummyLogprobs, res)
	}
}

func TestPublicCallOpenAIEnv(t *testing.T) {
	dummyLogprobs := []float64{7, 8}
	betterprompt.MockOpenAICompletion(dummyLogprobs)

	orig, present := setEnv("OPENAI_API_KEY", "public_env_key_test")
	defer func() {
		if present {
			os.Setenv("OPENAI_API_KEY", orig)
		} else {
			os.Unsetenv("OPENAI_API_KEY")
		}
	}()

	res, err := betterprompt.CallOpenAI("prompt string", "", "")
	if err != nil {
		t.Errorf("Unexpected error: %v", err)
	}
	if !equalFloatSlice(res, dummyLogprobs) {
		t.Errorf("Expected %v, got %v", dummyLogprobs, res)
	}
}

func TestPublicCalculatePerplexityDifferent(t *testing.T) {
	tokenLogprobs := []float64{-1, 0, 1, 2}
	expect := math.Exp(-sum(tokenLogprobs) / float64(len(tokenLogprobs)))
	actual := betterprompt.CalculatePerplexity(tokenLogprobs)
	if math.Abs(actual-expect) > 1e-8 {
		t.Errorf("Expected %v, got %v", expect, actual)
	}
}

File: RepoTransBench/target_projects/Python/Java/stjordanis_betterprompt/src/test/java/com/betterprompt/public_tests/PublicBetterPromptTest.java
package com.betterprompt.public_tests;

import com.betterprompt.BetterPrompt;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import org.mockito.MockedStatic;
import org.mockito.Mockito;

import java.util.*;

public class PublicBetterPromptTest {
    @Test
    public void testPublicCallOpenaiCustomModel() {
        List<Double> dummyLogprobs = Arrays.asList(-0.1, 2.5, -4.3);
        class DummyCompletion extends BetterPrompt.OpenAI.Completion {
            @Override
            public Map<String, Object> create(Object... args) {
                Map<String, Object> logprobs = new HashMap<>();
                logprobs.put("token_logprobs", dummyLogprobs);
                Map<String, Object> choice = new HashMap<>();
                choice.put("logprobs", logprobs);
                Map<String, Object> completion = new HashMap<>();
                completion.put("choices", Collections.singletonList(choice));
                return completion;
            }
            @Override
            public Map<String, Object> create(String prompt, String model, String apiKey) {
                assertEquals("public-model", model);
                return create();
            }
        }
        BetterPrompt.OpenAI.Completion = new DummyCompletion();
        List<Double> res = BetterPrompt.callOpenAI("sample prompt here", "public-model", "anypublickey");
        assertEquals(dummyLogprobs, res);
    }

    @Test
    public void testPublicCallOpenaiEnv() {
        List<Double> dummyLogprobs = Arrays.asList(7.0, 8.0);
        class DummyCompletion extends BetterPrompt.OpenAI.Completion {
            @Override
            public Map<String, Object> create(Object... args) {
                Map<String, Object> logprobs = new HashMap<>();
                logprobs.put("token_logprobs", dummyLogprobs);
                Map<String, Object> choice = new HashMap<>();
                choice.put("logprobs", logprobs);
                Map<String, Object> completion = new HashMap<>();
                completion.put("choices", Collections.singletonList(choice));
                return completion;
            }
            @Override
            public Map<String, Object> create(String prompt, String model, String apiKey) {
                return create();
            }
        }
        BetterPrompt.OpenAI.Completion = new DummyCompletion();
        try (MockedStatic<System> sysMock = Mockito.mockStatic(System.class)) {
            sysMock.when(() -> System.getenv("OPENAI_API_KEY")).thenReturn("public_env_key_test");
            List<Double> res = BetterPrompt.callOpenAI("prompt string");
            assertEquals(dummyLogprobs, res);
        }
    }

    @Test
    public void testPublicCalculatePerplexityDifferent() {
        List<Double> tokenLogprobs = Arrays.asList(-1.0, 0.0, 1.0, 2.0);
        double expected = Math.exp(-(-1.0 + 0.0 + 1.0 + 2.0) / 4.0);
        double actual = BetterPrompt.calculatePerplexity(tokenLogprobs);
        assertTrue(Math.abs(actual - expected) < 1e-8);
    }
}